cli/cmd/logs.go

```diff
@@ -0,0 +1,195 @@
+package cmd
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"regexp"
+	"text/template"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	"github.com/wercker/stern/stern"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+)
+
+//This code replicates most of the functionality in https://github.com/wercker/stern/blob/master/cmd/cli.go
+type logCmdOpts struct {
+	clientset *kubernetes.Clientset
+	*stern.Config
+}
+
+type logFlags struct {
+	containerFilter string
+	namespace       string
+	podFilter       string
+	containerState  string
+	labelSelector   string
+	sinceSeconds    time.Duration
+	tail            int64
+	timestamps      bool
+	follow          bool
+}
+
+func (c *logFlags) NewSternConfig() (*stern.Config, error) {
```

Comment:
I don't think this func needs to be exported publicly for use outside of the `cli` package. Also, I'd typically call a func that instantiates a struct `new*`, but since this one already has a different type of struct as its receiver, I'd call it `toSternConfig`.